fairscale/nn/checkpoint/checkpoint_activations.py

```diff
@@ -157,10 +157,16 @@ def checkpoint_wrapper(
 def _checkpointed_forward(
     original_forward: Any, weak_self: Any, offload_to_cpu: bool, *args: Any, **kwargs: Any
 ) -> Any:
+    module = weak_self()
+
+    # If in eval mode, just use the original `.forward()` method.
+    if not module.training:
```

Comment:
on line 236, we do check if grad is enabled. Can you share a test or the error you are running into?

Reply:
Updated